mlir/lib/Dialect/Bufferization/Transforms/FuncBufferizableOpInterfaceImpl.cpp

```diff
@@ -392,36 +393,45 @@ struct FuncOpInterface
     auto funcOp = cast<FuncOp>(op);
     FunctionType funcType = funcOp.getFunctionType();
 
-    // Construct the bufferized function type.
+    // Construct the bufferized function type. Compute the argument types.
```

Comment:
May be totally delete the `//Construct the bufferized function type` from here. The 'for loop' is about argType.
And then in line 421 you do have `// Compute the new function type.`